File: app/src/test/java/com/fractalwrench/crazycats/image/detail/DetailPresenterSuccessTest.java
package com.fractalwrench.crazycats.image.detail;


import com.fractalwrench.crazycats.mocks.MockImageDetailView;
import com.fractalwrench.crazycats.mocks.TestDependencies;

import org.junit.After;
import org.junit.Before;
import org.junit.Test;

import java.io.IOException;

import static org.junit.Assert.assertNotNull;
import static org.junit.Assert.assertNull;

public class DetailPresenterSuccessTest {

    private ImageDetailPresenter presenter;
    private MockImageDetailView detailView;

    @Before
    public void setup() throws IOException {
        presenter = new ImageDetailPresenter(TestDependencies.mockSuccessRepository());
        detailView = new MockImageDetailView();
    }

    @After
    public void tearDown() {
        presenter.stop();
    }

    /**
     * Ensures that the view is notified when the presenter fetches data
     */
    @Test
    public void checkPostListFailure() {
        assertNull(detailView.imageData);
        assertNull(detailView.errorMessage);

        presenter.start(detailView);
        assertNotNull(detailView.imageData);
        assertNull(detailView.errorMessage);
    }

}

File: app/src/test/java/com/fractalwrench/crazycats/image/detail/DetailPresenterSuccessTest.kt
package com.fractalwrench.crazycats.image.detail


import com.fractalwrench.crazycats.mocks.MockImageDetailView
import com.fractalwrench.crazycats.mocks.TestDependencies
import org.junit.After
import org.junit.Assert.assertNotNull
import org.junit.Assert.assertNull
import org.junit.Before
import org.junit.Test
import java.io.IOException

class DetailPresenterSuccessTest {

    private var presenter: ImageDetailPresenter? = null
    private var detailView: MockImageDetailView? = null

    @Before
    @Throws(IOException::class)
    fun setup() {
        presenter = ImageDetailPresenter(TestDependencies.mockSuccessRepository())
        presenter?.setId("")
        detailView = MockImageDetailView()
    }

    @After
    fun tearDown() {
        presenter!!.stop()
    }

    /**
     * Ensures that the view is notified when the presenter fetches data
     */
    @Test
    fun checkPostListFailure() {
        assertNull(detailView!!.imageData)
        assertNull(detailView!!.errorMessage)

        presenter!!.start(detailView!!)
        assertNotNull(detailView!!.imageData)
        assertNull(detailView!!.errorMessage)
    }

}
